adapters/repos/db/vector/hnsw/delete.go

```diff
@@ -750,25 +761,26 @@ func (h *hnsw) addTombstone(ids ...uint64) error {
 func (h *hnsw) removeTombstonesAndNodes(deleteList helpers.AllowList, breakCleanUpTombstonedNodes breakCleanUpTombstonedNodesFunc) (ok bool, err error) {
 	it := deleteList.Iterator()
 	for id, ok := it.Next(); ok; id, ok = it.Next() {
+		if breakCleanUpTombstonedNodes() {
+			return false, nil
+		}
 		h.metrics.RemoveTombstone()
 		h.tombstoneLock.Lock()
 		delete(h.tombstones, id)
 		h.tombstoneLock.Unlock()
 
 		h.resetLock.Lock()
-		if !breakCleanUpTombstonedNodes() {
-			h.shardedNodeLocks.Lock(id)
-			h.nodes[id] = nil
-			h.shardedNodeLocks.Unlock(id)
-			if h.compressed.Load() {
-				h.compressor.Delete(context.TODO(), id)
-			} else {
-				h.cache.Delete(context.TODO(), id)
-			}
-			if err := h.commitLog.DeleteNode(id); err != nil {
-				h.resetLock.Unlock()
-				return false, err
-			}
+		h.shardedNodeLocks.Lock(id)
+		h.nodes[id] = nil
+		h.shardedNodeLocks.Unlock(id)
+		if h.compressed.Load() {
+			h.compressor.Delete(context.TODO(), id)
```

Comment:
When you step into `compressor.Delete`, it looks like this method has the opportunity to fail, as it's essentially just wrapping `Bucket.Delete`. 

Is it possible that swallowing this error can be leading to issues? Maybe it's failing silently sometimes.

If quantization is enabled, and a deletion of the compressed vectors bucket fails, does it make sense to continue to delete the entry from the commitlog?

If it does, we should at least log the error IMO. If not, the error should be caught and returned so that the commitlog doesn't drift from the vector bucket contents

Reply:
I agree we should at least log the error. I think it does make sense to keep going with the logic at least until we know why such errors will occur in the first place. Not sure it is related to the problem we are solving with the PR yet since this would imply we have nodes in the cache that are not in the objects store, but still will not justify why those objects are in the graph. But I think a good first step is to log and if we have more information, we try to understand if/why a problem is happening.
Thanks @parkerduckworth.

Reply:
Addressed in this [commit](https://github.com/weaviate/weaviate/pull/5463/commits/9f40b787586b15953c5e5ab0b1c77c37ada583ae).

Reply:
Thanks, sounds good